packages/esm-patient-search-app/src/patient-search-icon/patient-search-icon.component.tsx

```diff
@@ -1,47 +1,163 @@
+import { useTranslation } from 'react-i18next';
+import { Button } from 'carbon-components-react';
+import debounce from 'lodash-es/debounce';
+import { useLayoutType } from '@openmrs/esm-framework';
+import { performPatientSearch } from '../patient-search/patient-search.resource';
+import isEmpty from 'lodash-es/isEmpty';
+import { SearchedPatient } from '../types';
 
 interface PatientSearchLaunchProps {}
 
+enum ActionTypes {
+  searching = 'searching',
+  resolved = 'resolved',
+  error = 'error',
+  idle = 'idle',
+}
+interface Searching {
+  type: ActionTypes.searching;
+  payload: Array<SearchedPatient>;
+}
+
+interface Error {
+  type: ActionTypes.error;
+  payload: Error;
+}
+interface Resolved {
+  type: ActionTypes.resolved;
+  payload: Array<SearchedPatient>;
+}
+
+interface Idle {
+  type: ActionTypes.idle;
+  payload: Array<SearchedPatient>;
+}
+
+type Action = Searching | Error | Resolved | Idle;
+
+interface PatientSearch {
+  status: 'searching' | 'resolved' | 'error' | 'idle';
+  searchResults: Array<SearchedPatient>;
+}
+
+function reducer(state: PatientSearch, action: Action): PatientSearch {
+  switch (action.type) {
+    case ActionTypes.resolved:
+      return { status: 'resolved', searchResults: action.payload };
+    case ActionTypes.searching:
+      return { status: 'searching', searchResults: action.payload };
+    case ActionTypes.error:
+      return { ...state, status: 'error' };
+    case ActionTypes.idle:
+      return { status: 'idle', searchResults: action.payload };
+  }
+}
+
+const searchTimeout = 300;
+const customRepresentation =
+  'custom:(patientId,uuid,identifiers,display,' +
+  'patientIdentifier:(uuid,identifier),' +
+  'person:(gender,age,birthdate,birthdateEstimated,personName,display),' +
+  'attributes:(value,attributeType:(name)))';
+
 const PatientSearchLaunch: React.FC<PatientSearchLaunchProps> = () => {
+  const { t } = useTranslation();
+  const layout = useLayoutType();
   const [open, setOpen] = useState<boolean>(false);
+  const [searchTerm, setSearchTerm] = useState<string>();
+  const initialState: PatientSearch = { status: 'idle', searchResults: [] };
+  const [{ searchResults, status }, dispatch] = useReducer(reducer, initialState);
 
-  const togglePatientSearch = useCallback(() => {
-    setOpen((prevState) => !prevState);
+  const performSearch = () => {
+    const ac = new AbortController();
+    if (searchTerm) {
+      dispatch({ type: ActionTypes.searching, payload: [] });
+      performPatientSearch(searchTerm, customRepresentation).then(
+        ({ data }) => {
+          const results: Array<SearchedPatient> = data.results.map((res, i) => ({
+            ...res,
+            index: i + 1,
+          }));
+          dispatch({ type: ActionTypes.resolved, payload: results });
+        },
+        (error) => {
+          dispatch({ type: ActionTypes.error, payload: error });
+        },
+      );
+    }
+    return () => ac.abort();
```

Comment:
Why is a function returned here? Do you intend to use this in a side-effect hook? Right now its not.